### PR TITLE
Adding follow me option to qground control

### PR DESCRIPTION
**Pull request by catch-twenty-two**
Adds new follow me class that sends motion information to follow me enabled vehicles over mavlink

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@catch-twenty-two Looking good so far. I have added the MAVLink proposal here: https://github.com/mavlink/mavlink/issues/509 and will execute on it timely to you're not blocked.

**Comment by LorenzMeier** [comment]
Can you rebase? A recent push to master probably led to a conflict.

**Comment by DonLakeFlyer** [comment]
Looks like a great so far, thanks. Only a few comments:
- Update copyrights to 2016
- Since you've already made FollowMe into a Tool it doesn't really need to be connected to MultiVehicleManager. It can just live on it's own can't it?
- FollowMe should only be sent to Vehicles which have follow me turned on shouldn't it? Instead of sending to all Vehicles?

**Comment by catch-twenty-two** [comment]
- Update copyrights to 2016
will do
- Since you've already made FollowMe into a Tool it doesn't really need to be connected to MultiVehicleManager. It can just live on it's own can't it?
Yep you are right, ill get rid of references to it here.
- FollowMe should only be sent to Vehicles which have follow me turned on shouldn't it? Instead of sending to all Vehicles?
I use a reference to the  vehicle manager and iterate through the currently connected vehicles in the class and see which ones are set to "follow me".  These are the only ones that get the follow me message via mavlink, all the rest are ignored. Let me know if there is a better way to identify whether or not something is set to a certain mode. 
https://github.com/mavlink/qgroundcontrol/pull/2817/files#diff-96fd8ff430a1d0cb5e35f7388ea390a1R83

**Comment by catch-twenty-two** [comment]
- Can you rebase? A recent push to master probably led to a conflict.
will do

**Comment by DonLakeFlyer** [comment]
Sorry missed the follow me on vehicle thing.

**Comment by LorenzMeier** [comment]
@catch-twenty-two I just updated the MAVLink submodule. Please rebase on master.

File: src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc [hunk 1]
```diff
@@ -85,6 +86,7 @@ static const struct Modes2Name rgModes2Name[] = {
     { PX4_CUSTOM_MAIN_MODE_AUTO,        PX4_CUSTOM_SUB_MODE_AUTO_RTL,       "Auto: Return To Land",   true },
     { PX4_CUSTOM_MAIN_MODE_AUTO,        PX4_CUSTOM_SUB_MODE_AUTO_LAND,      "Auto: Landing",          false },
     { PX4_CUSTOM_MAIN_MODE_AUTO,        PX4_CUSTOM_SUB_MODE_AUTO_RTGS,      "Return, Link Loss", false },
+    { PX4_CUSTOM_MAIN_MODE_AUTO,        PX4_CUSTOM_SUB_MODE_AUTO_FOLLOW_TARGET,  "Auto: Follow Me",        true },
```
Comment (LorenzMeier) [inline src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc]: Please rename os just "Follow Me"

Comment (catch-twenty-two) [inline src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc]: Will do

**Comment by catch-twenty-two** [comment]
> @catch-twenty-two I just updated the MAVLink submodule. Please rebase on master.

Awesome, thanks will do

**Comment by LorenzMeier** [comment]
@catch-twenty-two Could you rebase and fix all remaining CI issues? That would allow us to bring this in.

**Comment by LorenzMeier** [comment]
@DonLakeFlyer Any comments?

**Comment by DonLakeFlyer** [comment]
Haven't had time to look at it again. Give me a day.

**Comment by catch-twenty-two** [comment]
@LorenzMeier will do, I'm going to do one more integration test tomorrow then call it good.

**Comment by DonLakeFlyer** [comment]
I'll take another code review pass once it's rebased and passing CI.

**Comment by DonLakeFlyer** [comment]
@catch-twenty-two I can close this pull right?

**Comment by catch-twenty-two** [comment]
> @catch-twenty-two I can close this pull right?

@DonLakeFlyer yep

**Pull request closed without merging**